examples/tests/test_00_consumer.py

```diff
@@ -0,0 +1,129 @@
+"""
+Test the consumer with Pact.
+
+This module tests the consumer defined in `src/consumer.py` against a mock
+provider. The mock provider is set up by Pact, and is used to ensure that the
+consumer is making the expected requests to the provider, and that the provider
+is responding with the expected responses. Once these interactions are
+validated, the contracts can be published to a Pact Broker. The contracts can
+then be used to validate the provider's interactions.
+"""
+
+from __future__ import annotations
+
+import logging
+from http import HTTPStatus
+from typing import TYPE_CHECKING, Any, Generator
+
+import pytest
+import requests
+from examples.src.consumer import User, UserConsumer
+from pact import Consumer, Format, Like, Provider
+from yarl import URL
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pact.pact import Pact
+
+log = logging.getLogger(__name__)
+
+MOCK_URL = URL("http://localhost:8080")
+
+
+@pytest.fixture()
+def user_consumer() -> UserConsumer:
+    """
+    Returns an instance of the UserConsumer class.
+
+    As we do not want to stand up all of the consumer's dependencies, we direct
```

Comment:
Maybe we could be more concrete here, with regards to `all` of the dependencies, and the dependencies for the UserConsumer class.

Making it clear that we are only testing our collaborater code

https://docs.pact.io/5-minute-getting-started-guide#scope-of-a-consumer-pact-test

Not the full `consumer` app.

Probably fine as is!

Reply:
I've expanded on this within this docstring, and the link you suggest at the top :)